iceoryx_hoofs/include/iceoryx_hoofs/internal/cxx/helplets.inl

```diff
@@ -48,34 +82,32 @@ inline bool isValidFileName(const string<StringCapacity>& name) noexcept
     }
```

Comment:
A few lines above there is this check
```cpp
const string<StringCapacity> currentDirectory(".");
const string<StringCapacity> parentDirectory("..");

if (name == currentDirectory || name == parentDirectory)
{
    return false;
}
```
Even the code which follows is repeated in `isValidPathEntry`. Is this intentional?

Reply:
A there are subtle differences. 
The check in `isValidPathEntry` returns true for the `{current|parent}Directory` and false in `isValidFileName` since `a/../b` is allowed and here `..` would be a path entry but `..` would be an invalid file name.

Reply:
How about 
```cpp
enum class CurrentAndParentDirectory {
    Rejected,
    Accepted,
};
```
`isValidFileName` would become to just
```cpp
return isValidPathEntry(name, CurrentAndParrentDirectory::Rejected);
```
and `isValidPathEntry` would have a
```cpp
if (name == currentDirectory || name == parentDirectory)
{
    return currentAndParentDirectoryHandling == CurrentAndParrentDirectory::Accepted;
}
```

Something like this would remove the code duplication and result in 100% MC/DC coverage for this function ;)